src/Microsoft.Health.Fhir.Shared.Api/Features/Security/SecurityProvider.cs

```diff
@@ -44,19 +52,20 @@ public void Build(ICapabilityStatementBuilder builder)
             {
                 builder.Apply(statement =>
                 {
+                    bool stu3 = _modelInfoProvider.Version.Equals(FhirSpecification.Stu3);
                     if (_securityConfiguration.EnableAadSmartOnFhirProxy)
                     {
-                        AddProxyOAuthSecurityService(statement, _urlResolver, RouteNames.AadSmartOnFhirProxyAuthorize, RouteNames.AadSmartOnFhirProxyToken);
+                        AddProxyOAuthSecurityService(statement, _urlResolver, RouteNames.AadSmartOnFhirProxyAuthorize, RouteNames.AadSmartOnFhirProxyToken, stu3);
                     }
                     else
                     {
-                        AddOAuthSecurityService(statement, _securityConfiguration.Authentication.Authority, _httpClientFactory, _logger);
+                        AddOAuthSecurityService(statement, _securityConfiguration.Authentication.Authority, _httpClientFactory, _logger, stu3);
                     }
                 });
             }
         }
 
-        private static void AddProxyOAuthSecurityService(ListedCapabilityStatement statement, IUrlResolver urlResolver, string authorizeRouteName, string tokenRouteName)
+        private static void AddProxyOAuthSecurityService(ListedCapabilityStatement statement, IUrlResolver urlResolver, string authorizeRouteName, string tokenRouteName, bool stu3Version)
```

Comment:
These don't need to be static, thoughts on just adding `_modelInfoProvider.Version.Equals(FhirSpecification.Stu3) ? Constants.RestfulSecurityServiceStu3OAuth : Constants.RestfulSecurityServiceOAuth` to a Property and just using that?

Reply:
Why not :) On somewhat related note, any chance you know why we do Equals for enums instead of plain == ?

Reply:
Lol, no I don't